### PR TITLE
KBV-363 Publish JWKSet of public keys

### PR DESCRIPTION
**Pull request by idrop**
## Proposed changes

Publish JWKSet of public keys so that clients can easily find how to encrypt for the CRI, and validate the signature of the Verifiable Credential.

The url for the JWKSet is available from an AWS API call like `aws cloudformation describe-stacks --stack-name .... `

A sample response is:
````
{
  "keys": [
    {
      "kty": "EC",
      "use": "SIGN_VERIFY",
      "crv": "P-256",
      "kid": "4ae92735-e00f-4abb-b853-fc834e8f8aa4",
      "x": "gh1ghVk_Q8TeTMRnypAFeX9eQHwH20R9zgakPjgZUX4",
      "y": "Z3Ee-z2CVanVrqqBZRLsjJAr0nm6AEr6cp7Acbo6HDA"
    },
    {
      "kty": "RSA",
      "e": "AQAB",
      "use": "ENCRYPT_DECRYPT",
      "kid": "7fd1524b-24a1-43b9-83ec-7b62ab24f9f0",
      "n": "lqb_O2ya9uCBHzJhv9EQjKjXgIdF1HI_G9mu5LUuKHJeksu7du-t0hSzSdF_ItBYap5JaYwMago7lgY4OHOULiNWDcVPyL1-KirITl93aK8Bce5uuZC9-9ZllxdeYgY2cNu5cTmaRVwWyEUsGVGiHBKKnFuw8xmpood-2BD9k6Sh5mkRrNrGydvv36nAfyuDazBg0M3rH7RctC9YC17b-vJxIoASqFY4c2Yqf5o4vst_B8i-iu7cVQnj3PUp96LaJISZ8JzW9EKWXnG3AJNdUUG3n11XDPx2j5MNOHgK2ZoeiNKegf4GB_oU3YcRD-oY8iprEXbWBWD19VxPcS15Xw"
    }
  ]
}
````
The url is available at a url like `https://${AddressApi}.execute-api.${AWS::Region}.amazonaws.com/${Environment}/.well-known/jwks.json` which is similar to that which [Cognito uses](https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-verifying-a-jwt.html) for verifying JWT signatures.

### Why did it change

Helps to run acceptance tests automatically as JWKs will not be known on ephemeral environments.

### Issue tracking

- [KBV-363](https://govukverify.atlassian.net/browse/KBV-363)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -56,10 +56,13 @@ subprojects {
 		gson
 		powertools
 		mockito
+		bouncycastle
 	}
 
 	dependencies {
 
+		bouncycastle 'org.bouncycastle:bcpkix-jdk15on:1.70'
```
Comment (idrop) [inline build.gradle:64]: bouncycastle is a runtime dependency

File: infrastructure/lambda/template.yaml [hunk 2]
```diff
@@ -395,6 +428,9 @@ Resources:
             Action:
               - "kms:Sign"
             Resource: '*'
+      Tags:
```
Comment (idrop) [inline infrastructure/lambda/template.yaml:431]: JWKs must have this tag to be considered to be part of published JWKSet

File: build.gradle (hunk 1 above)
Comment (KaraokeStu) [inline build.gradle:64]: Is bouncy castle actually needed? I can't see it referenced in any of the code (unless it is abstracted away by Nimbus)

Comment (idrop) [reply build.gradle]: Oops, it isn't. - I'm removing it

Comment (idrop) [reply build.gradle]: It isn't needed! Removed

**Review by KaraokeStu — APPROVED** [review]

**Pull request merged**